izpack-panel/src/main/java/com/izforge/izpack/panels/datacheck/DataCheckPanel.java

```diff
@@ -78,95 +68,38 @@ public DataCheckPanel(Panel panel, InstallerFrame parent, GUIInstallData install
         add(scrollText);
-
     }
 
     /**
-     * When the panel is made active, call the printDebugInfo method.
+     * When the panel is made active, display debug information.
      *
      * @see com.izforge.izpack.installer.gui.IzPanel#panelActivate()
      */
     public void panelActivate()
     {
-        printDebugInfo();
-    }
-
-    /**
-     * Get and return the list of pack names.
-     *
-     * @param packList
-     * @return String
-     */
-    private String getPackNames(List<Pack> packList)
-    {
-        String pStatus;
-        String sOutput = "";
-        Pack iPack;
-        for (int i = 0; i < packList.size(); i++)
-        {
-            iPack = packList.get(i);
-            if (installDataGUI.getSelectedPacks().indexOf(iPack) != -1)
-            {
-                pStatus = "Selected";
-            }
-            else
-            {
-                pStatus = "Unselected";
-            }
-            sOutput = sOutput + "\t" + i + ": " + iPack.getName() + " (" + pStatus + ")\n";
-        }
-        return sOutput;
-    }
-
-    /**
-     * Print list of variables names and value, as well as the list
-     * of packages and their status (selected or not).
-     */
-    private void printDebugInfo()
-    {
-        int i = 0;
-        String sInfo = "GUIInstallData Variables:\n";
-        System.out.println("------------------------Data Check Panel Instance " +
-                                   instanceNumber + "------------------------");
-        System.out.println("GUIInstallData Variables:");
-        Properties varList = installDataGUI.getVariables().getProperties();
-        String[] alphaName = new String[varList.size()];
-        Enumeration<String> varNames = (Enumeration<String>) varList.propertyNames();
-        while (varNames.hasMoreElements())
-        {
-            alphaName[i++] = varNames.nextElement();
-        }
-        java.util.Arrays.sort(alphaName);
-        for (i = 0; i < alphaName.length; i++)
-        {
-            sInfo = sInfo + "\tName: " + alphaName[i] + ", Value: " + varList.getProperty(alphaName[i]) + "\n";
-        }
-        sInfo = sInfo + "\nAvailable Packs: \n" + getPackNames(installDataGUI.getAllPacks()) + "\n";
-        System.out.println(sInfo);
-        staticText.setText(sInfo);
-    }
-
-    /**
-     * By nature, always true.
-     *
-     * @return True
-     */
-    public boolean isValidated()
-    {
-        return true;
+        String packNames = DataCheckCommon.getPackNames(installData);
+        String panelBodyText = DataCheckCommon.getInstallDataVariables(installData) + "\n" +
+                (packNames == null ? "" : (packNames + "\n")) +
+                DataCheckCommon.getConditions(installData);
+        String infoToLog = DataCheckCommon.getMainLabelWithDashes(instanceNumber, panelId) + "\n" +
+                DataCheckCommon.SUB_LABEL + "\n" +
+                panelBodyText;
+
+        System.out.println(infoToLog);
```

Comment:
@hiteshbosamiya Should this go to a logger instead of System.out? I know it was like that before ...

Reply:
Yes, better approach, changed it.